crisischeckin/Services/SMSMessageSender.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Services.Interfaces;
+using Twilio;
+
+namespace Services
+{
+    public class SMSMessageSender : IMessageSender
+    {
+        private readonly string _senderPhone;
+        private readonly Func<TwilioRestClient> _twilioClient;
+
+        public SMSMessageSender(Func<TwilioRestClient> twilioClient, string senderPhone)
+        {
+            _twilioClient = twilioClient;
+            _senderPhone = senderPhone;
+        }
+
+        public void SendMessage(Message message, IReadOnlyCollection<MessageRecipient> recipients, string senderDisplayName = null)
+        {
+            TwilioRestClient twilioClient = _twilioClient();
```

Comment:
The suggested changes were added.

On Sun, Aug 9, 2015 at 11:53 AM, Bill Wagner <notifications@github.com>
wrote:

> In crisischeckin/Services/SMSMessageSender.cs
> <https://github.com/HTBox/crisischeckin/pull/325#discussion_r36589740>:
>
> > +namespace Services
> > +{
> > +    public class SMSMessageSender : IMessageSender
> > +    {
> > +        private readonly string _senderPhone;
> > +        private readonly Func<TwilioRestClient> _twilioClient;
> > +
> > +        public SMSMessageSender(Func<TwilioRestClient> twilioClient, string senderPhone)
> > +        {
> > +            _twilioClient = twilioClient;
> > +            _senderPhone = senderPhone;
> > +        }
> > +
> > +        public void SendMessage(Message message, IReadOnlyCollection<MessageRecipient> recipients, string senderDisplayName = null)
> > +        {
> > +            TwilioRestClient twilioClient = _twilioClient();
>
> In this code, if we are in the development environment, instead of sending
> text, write a text file with the information that will help people make
> sure their code works (the text recipients, and the content of the message)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HTBox/crisischeckin/pull/325/files#r36589740>.
>



-- 

<http://about.me/jsucupira>

[image: Jonathas Sucupira on about.me]

Jonathas Sucupira
about.me/jsucupira
  <http://about.me/jsucupira>